operator/src/main/java/oracle/kubernetes/operator/DomainRecheck.java

```diff
@@ -207,12 +210,24 @@ Step startNamespaceSteps(String ns) {
     @Override
     public NextAction apply(Packet packet) {
       NamespaceStatus nss = domainNamespaces.getNamespaceStatus(ns);
-      if (fullRecheck || !nss.isNamespaceStarting().getAndSet(true)) {
+      if (fullRecheck) {
         return doNext(packet);
+      } else if (isNamespaceStarting(nss)) {
+        return doNext(addEventStep(), packet);
       } else {
         return doEnd(packet);
       }
     }
+
+    private Step addEventStep() {
+      return Step.chain(
+          EventHelper.createEventStep(new EventData(NAMESPACE_WATCHING_STARTED).namespace(ns).resourceName(ns)),
+          getNext());
+    }
+
+    private boolean isNamespaceStarting(NamespaceStatus nss) {
```

Comment:
Should this method be called isNamespaceNotStarting() since it checks for negative condition?

Reply:
it detects if the namespace's isStarting boolean has just been changed from false to true, indicating it is starting right now.

Reply:
changed it to isNamespaceStartingChangedToTrue to avoid confusion.